Update usage docu for newcomers (#1037)
nf-core/mag: docs/usage/new_to_mag.md

- [**Domain and research specific guidance**](#domain-and-research-specific-guidance): discusses research-domain specific options (for example, when targeting specific organisms, or types of DNA).

## Brief introduction to nf-core/mag

### What does nf-core/mag do

The primary aim of nf-core/mag is to generate metagenomic assembled genomes (MAGs), perform quality control, and evaluate the quality of each MAG.

A run of nf-core/mag without any customisation or additional parameters will:

- Preprocess input reads to
  - Remove adapters (short reads: `fastp`, long reads: `porechop_ABI`)
  - FASTQ-level quality filtering (long reads: `Filtlong`)
  - Remove Phi X (short-read) or lambda (long-read) sequences
  - Quality control reads (short reads: FastQC, long reads: `Nanoplot`)
- Assemble reads into contigs with
  - `MEGAHIT` and `SPAdes` if short reads are provided
  - `SPAdesHybrid` if short and long reads are provided
  - `Flye` and `metaMDBG` if long reads are provided
- Post-assembly tasks
  - Quality control by `QUAST` and `ALE`
  - Annotate assemblies with `prodigal`
- Group contigs

>   - By binning with `MetaBat2`, `MaxBin2`, `CONCOCT`, `COMEBin`, `MetaBinner`, `SemiBin2`

Reviewer: ```suggestion
  - By binning with `MetaBat2`, `MaxBin2`, `CONCOCT`, `COMEBin`, `MetaBinner` and `SemiBin2`
```

Before you start, if you are unfamiliar with executing nf-core pipelines, we highly recommend reading the central [Getting started with nf-core](https://nf-co.re/docs/get_started/nf-core) documentation.

To run nf-core/mag, at a minimum, you will need:

- Nextflow
- A software dependency and compute environment manager (typically Conda, Docker, or Singularity/Apptainer)
- Metagenomic FASTQ files (Illumina or Nanopore sequence data is directly supported)
- A UNIX, and ideally internet connected, computing infrastructure (for example, laptop, server, high performance computing cluster (HPC), or cloud)

You will then need to prepare a [samplesheet](../usage#samplesheet-input-file) specifying the names and paths to the FASTQ Files.

Finally to execute, you will run a typical Nextflow command, either with [parameters](https://nf-co.re/mag/parameters/) customising the run specified on the command line,

```bash
nextflow run nf-core/mag --input samplesheet.csv --outdir <OUTDIR> -profile docker --clip_tool adapterremoval --reads_minlength 25 <...>
```

or with parameters specified in a [Nextflow parameter file](https://docs.seqera.io/nextflow/cli#pipeline-parameters).

```bash

> nextflow run nf-core/mag --input samplesheet.csv --outdir <OUTDIR> -profile docker

Reviewer: ```suggestion
nextflow run nf-core/mag -params-file params.json -profile docker
```
And add a small example of a YAML?

```

### What is the output from nf-core/mag?

nf-core/mag produces many different output directories and files, depending on what parameters you set.

Typically, the primary output files you will want from a typical run:

- `multiqc/multiqc_report.html`: For overall run summary statistics
- `GenomeBinning/bin_summary.tsv`: Summary statistics with quality metrics for the bins
- `GenomeBinning/<binner name>/`: The bin FASTA files themselves

You can explore example output from real data on the dedicated [Results](https://nf-co.re/mag/results/) page.

However, you will need to explore output on a per-project basis, as the exact files you need will depend on your question.

> You can see detailed descriptions of all output files on the [Output](https://nf-co.re/mag/5.4.2/docs/output/) page.

Reviewer: ```suggestion
You can see detailed descriptions of all output files on the [Output](https://nf-co.re/mag/latest/docs/output/) page.
```

## Brief introduction to nf-core/mag

### What does nf-core/mag do

The primary aim of nf-core/mag is to generate metagenomic assembled genomes (MAGs), perform quality control, and evaluate the quality of each MAG.

A run of nf-core/mag without any customisation or additional parameters will:

- Preprocess input reads to
  - Remove adapters (short reads: `fastp`, long reads: `porechop_ABI`)
  - FASTQ-level quality filtering (long reads: `Filtlong`)
  - Remove Phi X (short-read) or lambda (long-read) sequences
  - Quality control reads (short reads: FastQC, long reads: `Nanoplot`)
- Assemble reads into contigs with
  - `MEGAHIT` and `SPAdes` if short reads are provided
  - `SPAdesHybrid` if short and long reads are provided
  - `Flye` and `metaMDBG` if long reads are provided
- Post-assembly tasks
  - Quality control by `QUAST` and `ALE`
  - Annotate assemblies with `prodigal`
- Group contigs
  - By binning with `MetaBat2`, `MaxBin2`, `CONCOCT`, `COMEBin`, `MetaBinner` and `SemiBin2`
- Post-binning tasks
  - Quality control bins with `QUAST` and `BUSCO`

>   - Annotate bins with `PROKKA` (for bacteria and archaea) and `MetaEuk` (for Eukaryotes)

Reviewer: ```suggestion
  - Annotate bins with `PROKKA` (for bacteria and archaea) and `MetaEuk` (for eukaryotes)
```

> - Liu, S., et al. (2025). Analysis of metagenomic data. Nature Reviews. Methods Primers, 5(1), 5. [doi:10.1038/s43586-024-00376-6](https://doi.org/10.1038/s43586-024-00376-6)
> - New, F. N., & Brito, I. L. (2020). What Is Metagenomics Teaching Us, and What Is Missed? Annual Review of Microbiology, 74, 117–135. [doi:10.1146/annurev-micro-012520-072314](https://doi.org/10.1146/annurev-micro-012520-072314)

The pipeline will download database files for you where necessary.
All other functionality must be activated using dedicated [parameters](https://nf-co.re/mag/parameters).

### How do I run nf-core/mag?

Before you start, if you are unfamiliar with executing nf-core pipelines, we highly recommend reading the central [Getting started with nf-core](https://nf-co.re/docs/get_started/nf-core) documentation.

To run nf-core/mag, at a minimum, you will need:

- Nextflow
- A software dependency and compute environment manager (typically Conda, Docker, or Singularity/Apptainer)
- Metagenomic FASTQ files (Illumina or Nanopore sequence data is directly supported)

> - A UNIX, and ideally internet connected, computing infrastructure (for example, laptop, server, high performance computing cluster (HPC), or cloud)

Reviewer: ```suggestion
- A UNIX, and ideally internet-connected, computing infrastructure (for example, laptop, server, high performance computing cluster (HPC), or cloud)
```

Reviewer: - UNIX is a fairly technical term and probaly newcomers won't be familiar with it, so it would be clearer to explicitly mention Linux and macOS instead. Users working with other UNIX variants (e.g., BSD) will likely understand that the instructions apply to their systems as well.
- This says "ideally internet-connected", but in practice it is mandatory for a default run, because databases, containers, and Conda dependencies are downloaded on the fly.
- I would also mention that some databases are quite large and are best downloaded in advance. This recommendation already appears in the taxonomic assignment section, but it may be worth highlighting here as well so users are aware of it before they start.

Therefore, despite the larger computational overhead, testing multiple options can improve results.

However, reducing computational burden and accelerating analysis may require tool selection.

The following table summarises available assemblers:

| Assembler      | Input              | Comment                                                                                                                          |
| -------------- | ------------------ | -------------------------------------------------------------------------------------------------------------------------------- |
| `MEGAHIT`      | Short reads        | Fast and memory-efficient. Produces competitive assemblies though occasionally with higher misassembly rates compared to SPAdes. |
| `SPAdes`       | Short reads        | Computationally demanding but produces high-quality assemblies with lower misassembly rates. Slower than MEGAHIT.                |
| `SPAdesHybrid` | Short & long reads | Slow and memory-intensive. Leverages both read types for improved assembly accuracy.                                             |

> | `FLYE`         | Long reads         | Slow and memory-intensive. Suitable for long-read assemblies but not optimised for speed.                                        |

Reviewer: ```suggestion
| `Flye`         | Long reads         | Slow and memory-intensive. Suitable for long-read assemblies but not optimised for speed.                                        |
```

Reviewer: This sounds like "Flye bad, MetaMDBG good". Maybe add something like "Flye is more suitable for noisy reads / low coverage metagenomes" and MetaMDBG "works better with HiFi data".

Cite (from MetaMDBG article, https://www.nature.com/articles/s41467-026-69760-y): However, hifiasm-meta and metaMDBG both underperform on ONT data, even with the latest base-calling accuracy improvements. As a result, metaFlye remains the only viable option for assembling ONT datasets.

Reviewer: Nevermind, I guess it's more complex than that. Because that article is from the update from metaMDBG that introduces the specialized ONT pipeline, that in theory should perform equal or better than metaFlye. Let's leave that as it is for now I guess.

Reviewer 2: I think those are still reasonable recommendations, i've towned down the slow/fast thing by adding `-er`

> | `MetaDBG`      | Long reads         | Fast and memory-efficient alternative to FLYE for long-read assembly.                                                            |

Reviewer: ```suggestion
| `MetaMDBG`      | Long reads         | Fast and memory-efficient alternative to Flye for long-read assembly.                                                            |
```

> When both short and long reads are available, consider running `SPAdesHybrid` and/or **long-read assembly with FLYE or MetaDBG**.

Reviewer: ```suggestion
When both short and long reads are available, consider running `SPAdesHybrid` and/or **long-read assembly with Flye or MetaDBG**.
```

**Recommendation**: select on a per-project basis.

### Which bin quality control tool should I use?

Bin quality control checks for quality of the assembled bins against a range of criteria such as contamination and completeness, for example, against the MIMAG reporting standard ([Bowers et al. 2017](https://doi.org/10.1038/nbt.3893)).

Bin quality control is performed by default in nf-core/mag using BUSCO.
BUSCO evaluates both prokaryotes and eukaryotes based on marker genes.

Alternatives include **CheckM** (marker-gene-based, prokaryote-only) and **CheckM2** (machine learning-based, prokaryote-only).
CheckM is well established and commonly used, but the newer CheckM2 version has a more recent database and accurate evaluation.

Changing the default is typically driven by the need for comparability with other studies.

Additional chimerism checks with GUNC can be enabled if desired.

**Recommendation**: select on a per-project basis - BUSCO for projects with a broader taxonomic target, CheckM/CheckM2 for prokaryote only targets.

### Which bin taxonomic assignment tool should I use?

Taxonomic assignment involves comparing bins for similarity to known genomes and other MAGs.

> **GTDBTk** classifies bins using specific marker genes and yields GTDB-based taxonomies. This approach requires bins of at least medium quality for accuracy. GTDB only supports bacteria and archaea.

Reviewer: ```suggestion
**GTDB-Tk** classifies bins using specific marker genes and yields GTDB-based taxonomies. This approach requires bins of at least medium quality for accuracy. GTDB only supports bacteria and archaea.
```

**CAT**, by contrast, uses all detectable genes to assign NCBI-based taxonomies. CAT only supports microbes.

> The choice between GTDBTk and CAT depends on the desired taxonomy framework or the completeness of bins.

Reviewer: ```suggestion
The choice between GTDB-Tk and CAT depends on the desired taxonomy framework or the completeness of bins.
```

> [!WARNING]
> All taxonomic assignment tools for bins in nf-core/mag requires very large reference databases (10-100 GBs)!

**Recommendation**: select on a per-project basis.

> ## What parameters should I change?

Reviewer: ```suggestion
### What parameters should I change?
```

To run nf-core/mag, at a minimum, you will need:

- Nextflow
- A software dependency and compute environment manager (typically Conda, Docker, or Singularity/Apptainer)
- Metagenomic FASTQ files (Illumina or Nanopore sequence data is directly supported)
- A UNIX-based (Linux, Mac OSX etc.) computing infrastructure (for example, laptop, server, high performance computing cluster (HPC), or cloud)
  - Internet connection is needed for automatic download of databases, however you can download these files yourself. See [parameters](https://nf-co.re/mag/parameters/) for more information.
  - You will need a large amount of hard drive space, as some of the the databases (>100 GB per database) and output can be very large

You will then need to prepare a [samplesheet](../usage#samplesheet-input-file) specifying the names and paths to the FASTQ Files.

Finally to execute, you will run a typical Nextflow command, either with [parameters](https://nf-co.re/mag/parameters/) customising the run specified on the command line,

```bash

> nextflow run nf-core/mag --input samplesheet.csv --outdir results/ -profile docker --clip_tool adapterremoval --reads_minlength 25 <...>

Author: Personally I would never run a pipeline without using `-r`. Additionally I always prepend by `NXF_VER=<version>` to make sure my analysis will also work when the next nextflow update hits and the pipeline isnt compatible (that happened not only recently with 26.x.x).
Any opinions?

Reviewer 2: Nope, I would agree!

Reviewer 2: About the pipeline version I mean, I'm less concerned about the NXF_VER though - people use different ways of managing their dependencies (And I don't want to complicate the example tooooo much)

> nextflow run nf-core/mag --input samplesheet.csv --outdir results/ -profile docker --clip_tool adapterremoval --reads_minlength 25 <...>

Reviewer 2: ```suggestion
nextflow run nf-core/mag -r 5.4.2  --input samplesheet.csv --outdir results/ -profile docker --clip_tool adapterremoval --reads_minlength 25 <...>
```

```

or with parameters specified in a [Nextflow parameter file](https://docs.seqera.io/nextflow/cli#pipeline-parameters).

```bash

> nextflow run nf-core/mag -params-file params.json -profile docker

Reviewer 2: ```suggestion
nextflow run nf-core/mag -r 5.4.2 -params-file params.json -profile docker
```

| `Flye`         | Long reads         | Slower and more memory-intensive. Suitable for long-read assemblies but not optimised for speed. In some cases may perform better when you have issues with noisy reads or low coverage |
| `MetaDBG`      | Long reads         | Faster and more memory-efficient alternative to Flye for long-read assembly.                                                                                                            |

When both short and long reads are available, consider running `SPAdesHybrid` and/or **long-read assembly with Flye or MetaDBG**.

With high-depth long reads, long-read assembly typically yields more coherent results ([Agustinho et al. 2024](https://doi.org/10.1038/s41592-024-02262-1)).
Short-read-first assembly performs better with high-depth short reads or low-quality long reads and produces more fragmented but higher-accuracy assemblies ([Overholt et al. 2020](https://doi.org/10.1111/1462-2920.15186), [Meyer et al. 2022](https://doi.org/10.1038/s41592-022-01431-4)).

> **Recommendation**: run as many assemblers as computationally feasible.

Author: I think it would be helpful to add here approximate RAM and time requirements. Same for binners. Maybe we can add numbers that we produce for the manuscrtipt review here? Alternatively, a student of mine is working on checking computational requirements and we could add that numbers when they are available.

Reviewer 2: I'm always loathe to do make such things outside a dedicated study (which I really REALLY want to see, but few do) as it can be so highly dependent on the data and complexity input... but yes I guess we could add it from the stuff @dialvarezs is adding.

Reviewer 2: I guess we can with that provide read counts, but I assume the CAMI stuff is static? And then what metric of metagenome complexity can we use?

Reviewer 2: More context: I had this problem with a taxprofiler paper we just submitted, the only two metrics we could really report on were RAM and harddrive usagee...

Number of CPUs you can choose, you can set to 1 or 1000 it depends on how much is available, and time also depends many different factors: how old the CPU is, what type it is, how busy the scheduler is, what the IO bandwidth is, which are all driven by the infrastructure not the pipeline/tools themselves

Author: I would add this to give an approximation, i.e. does the specific assembler need 1 hour or 1 day, what tool needs longer? Is the RAM 10GB, 100GB, or 1TB? Number of CPUs kept on default. And thats it. After all its for people who are new to this to give an idea what to expect. We have that questions occationally in slack afaik.
And also how computational requirements relate to each other. I assume that in many cases the RAM and time ratios between alternative tools hold true, e.g. when one needs 10GB RAM and another needs 100GB, then the latter will usually need 10x more than the former. And that would be helpful in itself as well, I think.

Author: But we can also postpone that and attempt to add it in a separate PR? No need to complicate things?

| `CONCOCT`    | Unsupervised Gaussian mixture model clustering with strong performance in multi-sample binning. Frequently complemented by other binners in pipelines.                   |
| `COMEBin`    | Deep learning-based binner optimised for contrastive multi-view learning. Shows strong performance on hybrid and long-read assemblies.                                   |
| `Metabinner` | Machine learning approach combining contig composition and coverage. Designed for improved accuracy in complex metagenomes.                                              |
| `Semibin2`   | Semi-supervised binner using pretrained models and marker genes. Performs competitively on diverse metagenome types including challenging datasets.                      |

All binners currently run exclusively with CPUs.
GPU-based execution should accelerate several binners considerably.

> [!NOTE]
> CONCOCT and COMEBin typically have long run times, and are not recommended for time sensitive projects.
> They can produce better quality bins in some cases.

**Recommendation**: run as many binners as computationally feasible.

### Should I refine my bins?

> Bin refinement tools aims to cross-compare outputs, and selects the highest quality 'version' of the same bin from across all binners.

Author: ```suggestion
Bin refinement tools aim to cross-compare outputs, and select the highest quality 'version' of the same bin from across all binners.
```
